Домашняя работа №2 (Четыркина Лена)

lenabatik opened the pull request:
Т.к. второе задание основывается на первом, поэтому для проверки я отправляю пока 1 задание.

lenabatik committed:
Домашняя работа №2 (Четыркина Лена)

Т.к. второе задание основывается на первом, поэтому для проверки я отправляю пока 1 задание.

lenabatik committed:
Update README.md

Другую ссылку поставила

nburunova commented on README.md:
```diff
@@ -1,13 +1,57 @@
+1. Сайт OneTwoTrip - это онлайн-сервис для путешествий, где можно найти и забронировать номер по специальной цене в любом из 750 000 отелей по всему миру, купить билеты более 800 авиакомпаний с вылетом по всему миру и по всем направлениям.
+
+2. Декомпозиция функционала сервиса (блоки и подблоки):
+    1.	Меню
```
Меню нам описывает список доступного функционала: крупный функционал - это бронирование отелей, покупка билетов авиа и жд, помельче - Поддержка, больше информативный, чем интерактивный - это описание программ лояльности и приложений для мобильных. Можешь убрать блок меню и вместо него перечислить именно функционал, который скрывается за меню

lenabatik replied:
@nburunova Описание программ лояльности и приложений для мобильных, думаю, можно отнести к подблоку 9.3 "Упорядоченные статьи по темам" 9-го блока "Служба поддержки". Я посмотрела, есть там такие статьи.

nburunova commented on README.md:
```diff
@@ -1,13 +1,57 @@
+        5.1.	 Отели
+        5.2.	 Авиабилеты
+        5.3.	 ЖД
+    6.	Поиск информации
```
поиск я бы тоже разделила на поиск отелей, ЖД, авиа билетов, так как параметры поиска там разные

nburunova commented on README.md:
```diff
@@ -1,13 +1,57 @@
+    6.	Поиск информации
+        6.1.	 расширенный поиск
+        6.2.	 фильтрация информации
+    7.	Представление информации
```
Лучшие предложени у них только на отели и на авиа отображаются. Это тоже надо уточнить в этом пункте. Ведь ждя ЖД нет лучших вариантов

nburunova commented on README.md:
```diff
@@ -1,13 +1,57 @@
+    4.	Настройки
+        4.1.	 Выбор языка
+        4.2.	 Выбор фона сайта
+    5.	Справочники
```
а что ты подразумеваешь под справочниками? Что с ними можно делать? Или что они делают с пользователем? :)

lenabatik replied:
@nburunova под справочниками я понимаю базу данных для хранения всей информации про отели, авиабилеты и жд билеты. Сайт же ведь откуда-то должен предоставлять пользователю необходимую информацию.
Или этот блок убрать?

nburunova replied:
ок, имеет право быть

lenabatik committed:
Исправление первого задания д/з №2

Исправила первое задание по замечаниям. Насчет справочников ответила в комментариях. Еще на карту MindMup добавила пару новых функций.

lenabatik committed:
Выполнены оба задания в д/з №2

Дополнила карту подробной декомпозицией выбранного блока для 2-го задания.
Проанализировала выбранный блок и на основе этого написала набор тест-кейсов

nburunova commented on README.md:
```diff
@@ -1,39 +1,119 @@
-б) создать тестовый набор, указав в названии свои ФИО
-- Вкладка Test Suites and Cases -> Add Test Suite, в поле Name ввести свои ФИО и названия сайта, для которого пишеите тесты, например "Бурунова Надежда Skyscanner.com" -> кликнуть Add test suite
+3. MindMap - обновленная ссылка: https://atlas.mindmup.com/2015/11/cc561e406b390133b27c376ff917947b/one_two_trip/index.html
+   вместе с подробной декомпозицией для блока "Поиск авиабилетов" для 2-го задания
```
Лена, еще есть вариант "Наборный" - когда можно сложный маршрут искать. Не отражен в майнд мапе

nburunova commented on README.md:
```diff
@@ -1,39 +1,119 @@
+имеются два переключателя:
+	«туда и обратно» 
+    «в одну сторону»
+    
```
вариант "Наборный" : https://gyazo.com/0137ff19b55de8024a1f07cac36be3f0 https://gyazo.com/9c8f623d813e08184fdce693a9d17a45

nburunova commented:
Лена, ты в тестах проверила основной функционал, молодец.
Но я бы еще добавила, пункты пронумерованы в порядке приоритета: 
1. тесты на перелеты внутри России, между заграничными городами
2. тест на отсутствие результата по фильтрации, мы ведь можем так нафильтровать, что ничего не будет в результате?
3. проверила случай, когда билеты не найдены - надо убедиться, что приложение ведет себя правильно (например, из Южно-Сахалинска до Моранбы самолеты не летают) или когда ищем перелёт из одного города в тот же, интересно, что будет :)
4. добавила немного не функциональных тестов, например на поля ввода, они достаточно интересные. Есть крутое автозаполнение :) Тесты на то, верно ли срабатывает автозаполенние? а если ввели город с опечатками так, что система не смогла его дополнить? а что если оставили поле пустым (например, случайно нажали Ентер и запустился поиск на пустых полях)?

lenabatik commented:
@nburunova Надя, первый пункт "тесты на перелеты внутри России, между заграничными городами" - это для наборного варианта? например, из Лондона в Новосиб, из Новосиба в Москву, из Москвы в Питер, а потом из Питера в Париж ? Примерно так?
2. Тест на отсутствие результата по фильтрации, мы ведь можем так нафильтровать, что ничего не будет в результате
По сути это будет негативный тест-кейс?
3. Ясно
4. Я видела крутое автозаполнение. Например, набираю в поле novosibirsk, не успею допечатать, как сама система его переименовывает в Новосибирск.
А еще я пробовала набирать само слово "город" :) Система его переделала на "Гавр"

nburunova commented:
Про 1. 
Лена,  я имела ввиду для "В одну сторону" и для "Туда и обратно", мне показалось, что ты не рассмотрела этот вариант. Но сейчас гляжу в тест рейл и вижу, что рассмотрела для Туда-Обратно и для "В одну сторону" для российских городов. Извини, наверное это я не доглядела!

nburunova commented:
про 2. Да, негативный, но кажется, он часто может происходить

nburunova commented:
Про 4. Можешь придумать на него несколько тест-кейсов! :)

lenabatik commented:
@nburunova 
1. Добавила на карту Наборный поиск.
2. Добавила новые тест-кейсы.
3. Из Южно-Сахалинска до Моранбы самолеты оказывается летают. Для 19 ноября было сообщение, что нет билетов либо перелетов. Но на 20 ноября система благополучно показала наличие рейсов, правда, с двумя пересадками. В общем, я выбрала Ачинск, это маленький городок под Красноярском. Там аэропорта точно нет.
4. И еще я не понимаю, почему при автозаполнении введенное слово "город" система переводит на "Гавр", а слово "Масква" не хочет исправлять. Это нормально или плохо?
5. Заметила при наборном поиске, что допустим я заполняю сразу 5 запросов, а потом передумала и хочу убрать 5-й запрос, названия городов стираются из полей нормально. А вот выбранная дата никак не хочет убираться из поля! Мне кажется, что это очень плохо. Это значит, что я должна обновить страницу и снова заполнить 4 запроса...
6. Да, теперь про фильтрацию. Я не знаю как писать тест-кейс на отсутствие результата по фильтрации. Короче, если из Лондона по России, то богатого выбора нет, отображаются только два предложения насчет Санкт-Петербурга и Москвы. Причем "на любое время". А если выбрать на ровно 10 дней, в ноябре результатов таких нет, в январе тоже. В декабре на 10 дней есть и Питер, и Москва. В общем, мне кажется, обычная фильтрация, в зависимости от ситуации, наличия таких предложений, продолжительности отдыха и т.д. Тем более комбинации фильтрации у меня учитываются в AllPairs.

lenabatik committed:
Обновила карту

Добавила вариант "Наборный"
Добавила дополнительные тест-кейсы

nburunova commented:
1. на самом деле странно :) я не знаю, как должно точно быть :) в таких непонятных ситуациях надо разговаривать и разработчиками и\или заказчиком. Можешь написать в техподдержку и задать вопрос.
2. я думаю, что это если не баг, то кандидат на улучшение. Тоже напиши в техподдержку.
3. про фильтрацию: никак не удается найти такой вариант, чтобы по результатам фильтрации ничего не нашлось?

nburunova commented:
можешь приложить табличку с AllPairs?

nburunova commented:
Блок "Дополнительные тест-кейсы" и "Автоматическая корректировка названия города (1) \ (2)" - лучше избегать таких неговорящих названий. Попробуй поправить? 
И в самом конце пустой блок "Доп. тест-кейсы"

lenabatik commented:
@nburunova 
1. Хорошо, напишу в техподдержку ;)
2. Исправила неговорящие названия в наборе  тест-кейсов.
Действительно, чтобы исправить например "Автоматическая корректировка названия города (1) \ (2)", для этого мне пришлось заглянуть в тесты, чтобы прочитать, что там да как... Да, неудобно.
3. Пустой блок хотела убрать, да забыла... Уже убрала.
4. А как приложить табличку с AllPairs? В каком формате надо? На всякий случай оставлю здесь ссылку на табличку...
http://vk.com/doc27256810_437064828
5. Про фильтрацию: есть такой вариант, где нет никаких результатов. Это например вот при такой комбинации:
откуда: Лондон
куда: по России
когда: в декабре
на сколько: на выходные

nburunova commented:
1. Проверка фильтра лучших предложений авиабилетов - вот ты написала ... и т.д. и т.п., тесты в этой секции - они все однотипные (всего должно быть 43 тест-кейса)  - так лучше не делать, можно было приложить табличку с тестами из allpairs и в тесте написать: выполнить все 43 кейса из документа и ссылка на док или вложение
2. про фильтрацию - ты не писала тесты для случая, когда поискав какие-то перелёты, мы можем отфильтровать результат, например, оставив только бизнес-класс или только прямые рейсы? https://gyazo.com/beead4e8ad7ba0ca7201ba37bb4177b3

lenabatik commented:
@nburunova 
1. Исправила проверку фильтра лучших предложений авиабилетов, все заменила на один тест и приложила табличку AllPairs.
Ссылка на тест https://projectt2015.testrail.net/index.php?/cases/view/36&group_by=cases:section_id&group_id=16&group_order=asc
2. "...тесты для случая, когда поискав какие-то перелёты, мы можем отфильтровать результат, например, оставив только бизнес-класс или только прямые рейсы..."
Я такие тесты не писала. Надо написать?

nburunova commented:
1. я подумала, что наверное нет. ты же писала тесты не на фильтрацию, а на поиск билетов.
   Но стоит сделать тест кейс, где не будет результата - крайний случай поиска билета для каждого типа поиска (в одну сторону, туда-обратно, наборный)

lenabatik commented:
@nburunova 
Я добавила по тест-кейсу для каждого типа поиска, а для наборного сделала два теста, где "проблемный" запрос в качестве первой либо второй позиции в поиске.
Чтобы не было никаких результатов, в качестве пункта я выбрала Дамаск (столицу Сирии), поскольку там сейчас идет война и поэтому гражданские самолеты туда не летают.
Ссылки:
тест (в одну сторону) https://projectt2015.testrail.net/index.php?/cases/view/490
тест (туда и обратно) https://projectt2015.testrail.net/index.php?/cases/view/491
тест (наборный, проблемный первый запрос) https://projectt2015.testrail.net/index.php?/cases/view/493
тест (наборный, проблемный второй запрос) https://projectt2015.testrail.net/index.php?/cases/view/492

nburunova commented:
:+1:  принято
